pb/event.go

```diff
@@ -89,11 +93,63 @@ func (e *PushEvent) Validate() error {
 	return nil
 }
 
-type RepositoryInfo = vcsurl.RepoInfo //TODO(mcuadros): improve repository references
+// RepositoryInfo containes information about a repository
+type RepositoryInfo struct {
+	CloneURL string
+	Host     string
+	FullName string
+	Owner    string
+	Name     string
+}
+
+// list of hosts we allow when parse string into RepositoryInfo
+var supportedHosts = map[string]bool{
+	"github.com":    true,
+	"gitlab.com":    true,
+	"bitbucket.org": true,
+}
+
+// ParseRepositoryInfo creates RepositoryInfo from a string
+func ParseRepositoryInfo(input string) (*RepositoryInfo, error) {
+	u, err := url.Parse(input)
+	if err != nil {
+		return nil, err
+	}
+
+	if u.Scheme == "" {
+		return ParseRepositoryInfo("https://" + input)
+	}
+
+	if u.Scheme != "https" {
+		return nil, fmt.Errorf("only https urls are supported")
+	}
+
+	if u.Host == "" {
+		return nil, fmt.Errorf("host can't be empty")
+	}
+
+	if _, ok := supportedHosts[u.Host]; !ok {
+		return nil, fmt.Errorf("host %s is not supported", u.Host)
+	}
+
+	fullName := strings.Trim(u.Path, "/")
+	parts := strings.Split(fullName, "/")
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("unsupported path %s", fullName)
+	}
+
+	return &RepositoryInfo{
+		CloneURL: u.String(),
+		Host:     u.Host,
+		FullName: fullName,
+		Owner:    parts[0],
+		Name:     parts[1],
```

Comment:
What about a url like `https://github.com/src-d/lookout.git`, will this not put `lookout.git` into name instead of just `lookout`?